### DISCUSSION TITLE
Simulate USB reset

### DISCUSSION THREAD

**Discussion by mwpmorris** [tttapa/Control-Surface#1021] [Q&A]
Hi, I am trying to programatically force my host device to restart the connection negotiation process with my board running Control Surface.

I'm using a 32u4 and have been experimenting with USBDevice.detach() and USBDevice.detach() but lose Control Surface functionality after re-attaching.

I suspect that I should be gracefully stopping Control Surface before the detach, then re-starting after the attach.

Can anyone suggest an approach for this stop/restart of Control Surface, or an alternative approach?

Many thanks!

**Comment by tttapa** [comment]
On the ATmega32U4, Control Surface uses the Arduino MIDIUSB library as its backend. I'm not aware of anything that could go wrong inside of Control Surface itself when the USB connection is lost, although you will encounter issues if `MidiUSB.read()` suddenly runs into an infinite loop, for example.

What's the code you're using? Does it work with a bare minimum MIDIUSB library example?

**Reply by mwpmorris** [reply]
Thanks Pieter, I figured out a solution to the problem. Based on a bit of searching, I'm fairly sure that USBDevice.detach() wasn't implemented in my version of Arduino Core. I managed to find the register and set/unset the bit so that the USB port is reset and re-registers the board with the host unit. No need to stop / resume Control Surface or anything like that.
